src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java

```diff
@@ -140,12 +181,366 @@ public void testPopulatingExecutorService() throws Exception {
             executorServiceField.setAccessible(true);
             assertTrue(executorServiceField.get(blackDuckServerConfig) instanceof NoThreadExecutorService);
 
+            assertNotNull(blackDuckServerConfigBuilder.getExecutorService());
             blackDuckServerConfigBuilder.setExecutorService(executorService);
+            assertEquals(executorService, blackDuckServerConfigBuilder.getExecutorService().orElse(null));
             blackDuckServerConfig = blackDuckServerConfigBuilder.build();
             assertNotNull(executorServiceField.get(blackDuckServerConfig));
         } finally {
+            assert executorService != null;
             executorService.shutdownNow();
         }
     }
 
+    @Test
+    public void testNewBuilder() {
```

Comment:
The method names of the tests are not super helpful. I think something like testNewBuilderIncludesAllKeys() would be better. I also didn't see a test for 'provides username password to newApiTokenBuilder() and it is not valid', ie the detect use case.